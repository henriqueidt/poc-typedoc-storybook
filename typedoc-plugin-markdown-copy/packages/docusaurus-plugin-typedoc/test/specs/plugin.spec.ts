import * as fs from 'fs-extra';
import * as tmp from 'tmp';
import typedocPlugin from '../../dist/plugin';

tmp.setGracefulCleanup();

async function bootstrap(tmpobj: tmp.DirResult, customOptions = {}) {
  const options = {
    id: 'default',
    // logger: 'none',
    entryPoints: ['../../stub-project/src/theme.ts'],
    tsconfig: '../../stub-project/tsconfig.json',
  } as any;

  const plugin = typedocPlugin(
    {
      siteDir: tmpobj.name,
      generatedFilesDir: '',
      siteConfig: {},
    } as any,
    {
      ...options,
      ...customOptions,
    },
  );
  return await plugin.loadContent();
}

describe(`Plugin:`, () => {
  describe(`(docs)`, () => {
    let tmpobj;
    beforeAll(async () => {
      tmpobj = tmp.dirSync();
      await bootstrap(tmpobj, { sidebar: { collapsed: false } });
    });
    test(`should render`, () => {
      const files = fs.readdirSync(tmpobj.name + '/docs/api');
      expect(files.sort((a, b) => a.localeCompare(b))).toMatchSnapshot();
    });

    test(`should write doc`, () => {
      const sidebar = fs.readFileSync(tmpobj.name + '/docs/api/index.md');
      expect(sidebar.toString()).toMatchSnapshot();
    });

    test(`should write category yaml`, () => {
      const sidebar = fs.readFileSync(
        tmpobj.name + '/docs/api/classes/_category_.yml',
      );
      expect(sidebar.toString()).toMatchSnapshot();
      const parentSidebar = fs.readFileSync(
        tmpobj.name + '/docs/api/_category_.yml',
      );
      expect(parentSidebar.toString()).toMatchSnapshot();
    });
  });
});
