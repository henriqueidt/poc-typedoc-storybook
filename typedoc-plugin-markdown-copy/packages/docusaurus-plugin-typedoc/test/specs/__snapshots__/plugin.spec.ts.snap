// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin: (docs) should render 1`] = `
[
  "_category_.yml",
  "classes",
  "enums",
  "index.md",
  "interfaces",
  "modules.md",
  "namespaces",
]
`;

exports[`Plugin: (docs) should write category yaml 1`] = `
"label: "Classes"
position: 3"
`;

exports[`Plugin: (docs) should write category yaml 2`] = `
"label: "API"
collapsed: false"
`;

exports[`Plugin: (docs) should write doc 1`] = `
"---
id: "index"
title: "typedoc-plugin-markdown"
sidebar_label: "Readme"
sidebar_position: 0
custom_edit_url: null
---

# MY README

> Some block quote

Some text
"
`;
